### PR TITLE
Suggestions to help a developer using this repo

### PR DESCRIPTION
**Pull request by rebeccapeltz**
I worked through the readme's and the code, and found some things that might improve getting started with this.  Most suggested changes are in the readme's and package.json's.  
There is currently a fail with the initial command to `npm install cloudinary` because the name of the package.json is also 'cloudinary`.  I added `cloudinary` to the dependencies in the root package.json and then just `nom install` in the readme instructions.  
I ran the tests and there are 
 492 passing (3m)
  8 failing
and that matched what I saw before making modifications.

### CODE REVIEW

File: samples/photo_album/package.json [hunk 1]
```diff
@@ -9,14 +9,16 @@
   "author": "",
   "license": "ISC",
   "dependencies": {
+    "blueimp-file-upload": "^10.2.0",
```
Comment (asisayag2) [inline samples/photo_album/package.json]: @rebeccapeltz , why are those changes needed?

File: samples/basic/basic.js [hunk 2]
```diff
@@ -145,3 +146,24 @@ function performTransformations() {
   console.log("> That's it. You can now open the URLs above in a browser");
   console.log("> and check out the generated images.");
 }
+
+//using async/await with cloudinary upload
```
Comment (asisayag2) [inline samples/basic/basic.js]: @rebeccapeltz , you added that method, but what's calling it?

File: package.json [hunk 3]
```diff
@@ -11,6 +11,7 @@
   },
   "main": "cloudinary.js",
   "dependencies": {
+    "cloudinary": "^1.16.0",
```
Comment (asisayag2) [inline package.json]: @rebeccapeltz , why is that needed?

File: package-lock.json [hunk 4]
```diff
@@ -0,0 +1,7998 @@
+{
+  "name": "cloudinary",
```
Comment (asisayag2) [inline package-lock.json]: @rebeccapeltz , please don't commit package-lock files

File: samples/photo_album/package.json (hunk 1 above)
Comment (rebeccapeltz) [reply samples/photo_album/package.json]: security audit  - when you npm install you get security errors - this can be automated for a repo in github but I just updated it manually - I'm attaching a picture showing it automated

auto-security

Comment (rebeccapeltz) [reply samples/photo_album/package.json]: If you're not using package-lock.json you can ignore the above.  I found that people were uncomfortable with code with npm security alerts in open source.

File: package-lock.json (hunk 4 above)
Comment (rebeccapeltz) [reply package-lock.json]: If not committing package-lock.json, should you add to .gitignore?   
If interested in letting github automate security updates here's an article from them https://help.github.com/en/github/managing-security-vulnerabilities/configuring-automated-security-updates

File: samples/basic/basic.js (hunk 2 above)
Comment (rebeccapeltz) [reply samples/basic/basic.js]: see in this function  the `await open(url) . `  ```async function uploadImage(filename) {
  try {
      const result = await cloudinary.uploader.upload(filename, { tags: 'basic_sample' });
      console.log(result)
      const url = await cloudinary.url(result.public_id, {
          width: 400,
          quality: 'auto',
          fetch_format: 'auto',
          secure: true
      });
      console.log(url)
      await open(url)
  }
  catch (err) {
      console.log(err)
  }
}```

File: package.json (hunk 3 above)
Comment (rebeccapeltz) [reply package.json]: manual update to latest

File: README.md [hunk 5]
```diff
@@ -17,7 +17,7 @@ For Node.js, Cloudinary provides an extension for simplifying the integration ev
 
 ## Setup ######################################################################
 
-npm install cloudinary
+npm install
```
Comment (patrick-tolosa) [inline README.md]: This should actually stay as `npm install cloudinary` as this readme file discusses how to install the project as a dependency in your own project.

running this command inside the cloudinary project will result in an error, but that the correct behavior.

Can you please fix this specifically so we can approve the PR?

Comment (rebeccapeltz) [reply README.md]: If you just want me to resolve this I can, but I'm wondering why you don't use  `npm i`.  Do you want the user to experience an error.  Why give them a pacakge.json if they can't isntall it per instructions?  Running `npm i` will isntall cloudinary without getting an error.  This is what I see when I run `npm install cloudinary`
rebeccapeltz@MacBook-Pro-2 cloudinary_npm % npm install cloudinary
npm ERR! code ENOSELF
npm ERR! Refusing to install package with name "cloudinary" under a package
npm ERR! also called "cloudinary". Did you name your project the same
npm ERR! as the dependency you're installing?
npm ERR! 
npm ERR! For more information, see:
npm ERR!     <https://docs.npmjs.com/cli/install#limitations-of-npms-install-algorithm>
see this link: https://docs.npmjs.com/cli/install#limitations-of-npms-install-algorithm

File: package.json (hunk 3 above)
Comment (rebeccapeltz) [reply package.json]: just noticed that when I merged upstream, it didn't pickup this change - that you had moved cloudinary out of dependencies - not sure why the package.json merge didn't work but I see that you've moved cloudinary to "main" so the npm install runs without error

File: README.md (hunk 5 above)
Comment (rebeccapeltz) [reply README.md]: just noticed that you moved cloudinary out of dependencies so `npm install cloudinary runs` - my upstream merge didn't pick that up for some reason

**Comment by rebeccapeltz** [comment]
Sorry if this held you up.  I think I had some trouble merging from upstream.  Maybe I'm not authorized.

File: samples/basic/basic.js (hunk 2 above)
Comment (rebeccapeltz) [reply samples/basic/basic.js]: Open is a function from https://www.npmjs.com/package/open . I see you are providing instructions to open in browser.  The open function opens in default browser.
